Translation/Extractor/File/FormExtractor.php

```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
-                    continue;
-                }
+                $domain = $item->value->value;
+            }
+        }
 
-                if ('choices' === $item->key->value) {
+        return $domain;
+    }
 
-                    //Checking for the choice_as_values in the same form item
-                    $choicesAsValues = false;
-                    foreach ($node->items as $choiceItem) {
-                        if ($choiceItem->key !== null && 'choices_as_values' === $choiceItem->key->value) {
-                            $choicesAsValues = ($choiceItem->value->name->parts[0] === 'true');
-                        }
-                    }
-
-                    foreach ($item->value->items as $sitem) {
-                        // If we have a choice as value that differ from the Symfony default strategy
-                        // we should invert the key and the value
-                        if (Kernel::VERSION_ID < 30000 && $choicesAsValues === true || Kernel::VERSION_ID >= 30000) {
-                            $newItem = clone $sitem;
-                            $newItem->key = $sitem->value;
-                            $newItem->value = $sitem->key;
-                            $sitem = $newItem;
-                        }
-                        $this->parseItem($sitem, $domain);
-                    }
-                } elseif ('attr' === $item->key->value && $item->value instanceof Node\Expr\Array_) {
-                    foreach ($item->value->items as $sitem) {
-                        if ('placeholder' == $sitem->key->value) {
-                            $this->parseItem($sitem, $domain);
-                        }
-                        if ('title' == $sitem->key->value) {
-                            $this->parseItem($sitem, $domain);
-                        }
-                    }
-                } elseif ('invalid_message' === $item->key->value) {
-                    $this->parseItem($item, 'validators');
-                } else {
-                    $this->parseItem($item, $domain);
-                }
+    /**
+     * @param Node $item
+     * @param $domain
+     * @return bool
+     */
+    public function parseEmptyValueNode(Node $item, $domain)
+    {
+        // Skip empty_value when false
+        if ($item->value instanceof Node\Expr\ConstFetch && $item->value->name instanceof Node\Name && 'false' === $item->value->name->parts[0]) {
+            return true;
+        }
+
+        // Parse when its value is an array of values
+        if ($item->value instanceof Node\Expr\Array_) {
+            foreach ($item->value->items as $subItem) {
+                $this->parseItem($subItem, $domain);
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param Node $item
+     * @param Node $node
+     * @param $domain
+     * @return bool
+     */
```

Comment:
Same here, it needs docs. What does this function do? Why? What does the return value mean?